src/main/java/com/yahoo/bullet/rest/service/QueryService.java

```diff
@@ -12,23 +12,19 @@
 import com.yahoo.bullet.pubsub.Publisher;
 import com.yahoo.bullet.pubsub.Subscriber;
 import com.yahoo.bullet.query.Query;
-import com.yahoo.bullet.rest.PubSubConfiguration.PubSubResponderList;
 import com.yahoo.bullet.rest.common.PublisherRandomPool;
 import com.yahoo.bullet.rest.common.Reader;
 import com.yahoo.bullet.rest.common.Utils;
 import com.yahoo.bullet.storage.StorageManager;
 import lombok.extern.slf4j.Slf4j;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.stereotype.Service;
 
 import javax.annotation.PreDestroy;
 import java.util.List;
 import java.util.Objects;
 import java.util.concurrent.CompletableFuture;
 import java.util.stream.Collectors;
 
-@Service @Slf4j
```

Comment:
It should still be a Service even if it's not Autowired

Reply:
Making it a service autowires it automatically, e.g. it complains that "int sleep" in the constructor is not auto-wirable even w/o Autowrired annotation on the constructor.